merlin-server/sql/merlin/functions/hasura/hasura_functions.sql

```diff
@@ -27,3 +27,58 @@ begin
 	  order by p.name, ps.start_offset desc;
 end
 $$;
+
+create function hasura_functions.restore_activity_directive_from_changelog(
+  _plan_id integer,
+  _activity_directive_id integer,
+  _revision integer,
+  hasura_session json)
+  returns setof activity_directive
```

Comment:
Hm, I wonder if this can be changed from `returns setof` to just `returns`.

Reply:
Looks like this does need to be `returns setof` when using `return query`